alpa/shard_parallel/compile_executable.py

```diff
@@ -49,101 +48,46 @@ def get_compute_key(fun: lu.WrappedFun, in_tree: PyTreeDef,
     return hash_key
 
 
-def shard_parallel_callable(
+def compile_shard_executable(
     fun: lu.WrappedFun,
     in_tree: PyTreeDef,
     out_tree_thunk: Callable,
     static_argnums: Sequence[int],
     donated_invars: Sequence[bool],
     batch_invars: Sequence[bool],
-    devices,
-    memory_budget_per_device: float,
-    *avals: Sequence[ShapedArray],
+    device_mesh: Union[PhysicalDeviceMesh, LogicalDeviceMesh],
+    num_micro_batches: Optional[int],
+    as_option: AutoShardingOption,
```

Comment:
`as` can also be interpreted as "auto stage". Can we use `auto_sharing_option` instead?

Reply:
This is under `shard_parallel`, so it should not be related to any pipeline stages.

Reply:
This abbreviation is used widely. For example, the following line includes both `default_as_option` and `auto_stage_option`, which is confusing.
https://github.com/alpa-projects/alpa/blob/60b5d9a0b86503679fc7085a8a9ff5227e09b1ab/alpa/pipeline_parallel/stage_construction.py#L272